codalab/rest/interpret.py

```diff
@@ -306,134 +314,86 @@ def resolve_interpreted_items(interpreted_items):
     appropriate information, replacing the 'interpreted' field in each item.
     The result can be serialized via JSON.
     """
-    def error_data(mode, message):
-        if mode == 'record' or mode == 'table':
-            return (('ERROR',), [{'ERROR': message}])
-        else:
-            return [message]
+    def error_data(item_index, message):
```

Comment:
Why make this mutable?  If you really need it to mutate, then rename to `set_error_data`.

Reply:
I got the impression that this entire function is mutating the contents of interpreted_items anyway, I can make a non-mutating version that creates a copy first and fills it in.